mystery-deckbuilder/Assets/Scripts/Encounter/EncounterScript.cs

```diff
@@ -90,39 +151,270 @@ public int GetCompliance()
+     * changes the text colour to show relation to base value
+     */
+    private void UpdateCards()
+    {
+        
+        foreach (KeyValuePair<GameObject, ConversationCard> i in handFrontend) //looks like an optimization problem, but N of handFrontend is max 5 and N of textFeilds is max 4
+        {
+            int[] modifyAmounts = (int[])ResolveFilters(i.Value); //[0] = patience modifier, [1] = compliance modifier
+
+            Text[] textFields = i.Key.GetComponentsInChildren<Text>();
+
+            foreach (Text j in textFields)
+            {
+                
+                if (j.CompareTag("Patience"))
+                {
+                    j.text = (i.Value.GetPatienceValue() + modifyAmounts[0]).ToString();
+                    if(i.Value.GetPatienceValue() + modifyAmounts[0] > i.Value.GetPatienceValue())
+                    {
+                        j.color = Color.green;
+                    }
+                    else if(i.Value.GetPatienceValue() + modifyAmounts[0] < i.Value.GetPatienceValue())
+                    {
+                        j.color = Color.red;
+                    }
+                    else
+                    {
+                        j.color = Color.black;
+                    }
+                    
+                }
+                else if (j.CompareTag("Compliance"))
+                {
+                    j.text = (i.Value.GetComplianceValue() + modifyAmounts[1]).ToString();
+                    if (i.Value.GetComplianceValue() + modifyAmounts[1] > i.Value.GetComplianceValue())
+                    {
+                        j.color = Color.green;
+                    }
+                    else if (i.Value.GetComplianceValue() + modifyAmounts[1] < i.Value.GetComplianceValue())
+                    {
+                        j.color = Color.red;
+                    }
+                    else
+                    {
+                        j.color = Color.black;
+                    }
+                }
+            }
+
         }
-        complianceBar.GetComponent<BarScript>().SetValue(_compliance);
     }
 
+    /** 
+     * card object passes the id of the static object that calculates it's effect and the durration of how long that effect lingers.
+     * "local" effects can just be duration 1 or something
+     * this implementation allows cards to pass their lingering effects without them disapearing when the card is destroyed
+     * 
+     */
+    public void AddFilter(int duration, int id)
+    {
+        List<int> filter = new List<int>() {duration, id };
+        filters.Add(filter);
+    }
+
+    private Array ResolveFilters(ConversationCard card)
+    {
+        
+        int[] modifyAmounts = new int[] {0, 0}; //[0] = patience modifier, [1] = compliance modifier
+
+        foreach (List<int> i in filters)
+        {
+            int[] tempArray = (int[])Filters.GetFilterByID(i[1], card.GetComplianceValue(), card.GetPatienceValue(), card.GetElement());
```

Comment:
For clarity, I don't see why you don't just pass card to the Filter.cs script and have the resolving logic there.